src/Analytics/UAExperimentData.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace PhpAb\Analytics;
+
+use Webmozart\Assert\Assert;
+
+class UAExperimentData
+{
+
+    /**
+     * @var array Of test identifiers and their variant's indexes
+     */
+    private $testsData = [];
+
+    /**
+     * @param string $testIdentifier It will look like "Qp0gahJ3RAO3DJ18b0XoUQ"
+     * @param int $variationIndex
+     * @return \PhpAb\Analytics\UAExperiments
+     * @throws InvalidArgumentException
+     */
+    public function onRegisterParticipation($testIdentifier, $variationIndex)
+    {
+        Assert::string($testIdentifier, 'Test identifier must be astring');
+
+        Assert::integer($variationIndex, 'Variation index must be integer');
+
+        Assert::greaterThan($variationIndex, -1, 'Variation index must be integer >= 0');
+
+        $this->testsData[$testIdentifier] = $variationIndex;
+
+        return $this;
```

Comment:
If this class will could only be used via Event system then there's no need, otherwise I don't see why not to have a fluent interface